synapseclient/client.py

```diff
@@ -522,7 +522,7 @@ async def close_connection() -> None:
         self._requests_session_async_synapse.update(
             {
                 asyncio_event_loop: httpx.AsyncClient(
-                    limits=httpx.Limits(max_connections=25),
+                    limits=httpx.Limits(max_connections=5),
```

Comment:
I am purposely lowering this limit. When I was testing out this logic, and some download logic, I saw a significant amount of rate limit messages. Significantly fewer after lowering this amount. This will help in the situations in this new logic where we are doing a `asyncio.gather` call. This call attempts to execute ALL asyncio tasks at once, however, it would cause more rate limit issues. By lowering the connections at the HTTP client it gives us a central location to control the "Funnel" to requests going to Synapse rest APIs.